compiler/src/dmd/cpreprocess.d

```diff
@@ -40,13 +41,12 @@ version (Windows) version = runPreprocessor;
  * Params:
  *      csrcfile = C file to be preprocessed, with .c or .h extension
  *      loc = The source location where preprocess is requested from
- *      ifile = set to true if an output file was written
  *      defines = buffer to append any `#define` and `#undef` lines encountered to
  * Result:
- *      filename of preprocessed output
+ *      the text of the preprocessed file
  */
 extern (C++)
-FileName preprocess(FileName csrcfile, ref const Loc loc, out bool ifile, OutBuffer* defines)
+Buffer preprocess(FileName csrcfile, ref const Loc loc, ref OutBuffer defines)
```

Comment:
Thanks for writing up an alternative so I quickly, the only problem I spot here is the return of value.

Buffer [has a dtor](https://github.com/dlang/dmd/blob/79b2813c53cf26de19d3e37875136a410aebaf7f/compiler/src/dmd/root/file.d#L40) so it returns by (invisible) reference ...

Reply:
changed it to DArray